src/particles/ReferenceParticle.H

```diff
@@ -39,6 +40,9 @@ namespace impactx
         amrex::ParticleReal mass = 0.0; ///< reference rest mass, in kg
         amrex::ParticleReal charge = 0.0; ///< reference charge, in C
 
+        amrex::ParticleReal sedge = 0.0;  ///< value of s at entrance of the current beamline element
+        amrex::Array2D<amrex::ParticleReal, 1, 6, 1, 6> map; ///< linearized map
```

Comment:
I am ok to have these 1-based, close to the mathematical formula.
But I think the compiler might be able to simplify a few integer operations when we access the arrays at runtime (offsets and then modulo operations) if we had it starting at zero.

cc @WeiqunZhang ?

Reply:
I agree 1-based indexing should have the same speed as 0-based because it's a known constant at compile time.